src/backend/utils/indexer.js

```diff
@@ -48,15 +54,16 @@ const deletePost = async (postId) => {
 /**
  * Searches text in elasticsearch
  * @param textToSearch
+ * @param filter
  * @return all the results matching the passed text
  */
-const search = async (textToSearch) => {
+const search = async (textToSearch, filter = 'post') => {
   const query = {
     query: {
       simple_query_string: {
         query: textToSearch,
         default_operator: 'and',
-        fields: ['text'],
+        fields: filter === 'post' ? ['text', 'title', 'author'] : ['author'],
```

Comment:
We might add more filters later (e.g., by date, or just title, ...) so let's make a function for this instead:

```js
const createFieldsFromFilter = (filter) => {
  switch(filter) {
    case 'author':
      return ['author']
    case 'post':
    default:
      return ['text', 'title'];
    }
}
```

And then you can call it here:

```
fields: createFieldsFromFilter(filter),
```